ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStore.java

```diff
@@ -28,6 +28,8 @@
 
   Optional<StructuredDataSource> getSourceForTopic(String ksqlTopicName);
 
+  Optional<StructuredDataSource> getSourceForKafkaTopic(String kafkaTopicName);
```

Comment:
We have a one-to- many relation between kafka topic and StructuredDataSource. For instance, you can declare multiple KSQL streams on the same kafka topic. You need to return a collection(maybe set or list) here.

Reply:
Good point. Switched to return a list instead.